### PR TITLE
Ruff: add and fix EM rules

### PR DESCRIPTION
**Pull request by kiblik**
Fix  https://docs.astral.sh/ruff/rules/#flake8-errmsg-em / https://pypi.org/project/flake8-errmsg/ rules

### CODE REVIEW

**Review by Maffooch — APPROVED** [review]

File: dojo/tools/api_sonarqube/api_client.py [hunk 1]
```diff
@@ -38,9 +38,8 @@ def __init__(self, tool_config):
                 entry in supported_issue_types for entry in split_issue_types
             )
             if not all_clean:
-                raise Exception(
-                    f"Deteced unsupported issue type! Supported types are {', '.join(supported_issue_types)}"
-                )
+                msg = f"Deteced unsupported issue type! Supported types are {', '.join(supported_issue_types)}"
```
Comment (cneill) [inline dojo/tools/api_sonarqube/api_client.py:41]: While we're modifying these messages, might as well clean up typos...

```suggestion
                msg = f"Detected unsupported issue type! Supported types are {', '.join(supported_issue_types)}"
```

File: dojo/tools/factory.py [hunk 2]
```diff
@@ -25,21 +25,24 @@ def register_parser(scan_type, parser):
...
+        msg = f"Try to register an existing parser '{scan_type}'"
+        raise ValueError(msg)
     PARSERS[scan_type] = parser
 
 
 def get_parser(scan_type):
     """Return a parser by the scan type"""
     if scan_type not in PARSERS:
-        raise ValueError(f"Parser '{scan_type}' does not exists")
+        msg = f"Parser '{scan_type}' does not exists"
```
Comment (cneill) [inline dojo/tools/factory.py:36]: Just another typo

```suggestion
        msg = f"Parser '{scan_type}' does not exist"
```

File: dojo/tools/threagile/parser.py [hunk 3]
```diff
@@ -73,15 +73,17 @@ def get_findings(self, file, test):
...
         if not tree:
             return list()
         findings = []
         for item in tree:
             for field in self.REQUIRED_FIELDS:
                 if field not in item.keys():
+                    msg = f"Invalid ThreAgile risks file, missing field {field}"
                     raise ValueError(
-                        f"Invalid ThreAgile risks file, missing field {field}")
+                        msg)
```
Comment (cneill) [inline dojo/tools/threagile/parser.py:86]: No need for a newline here...

File: dojo/tools/yarn_audit/parser.py [hunk 4]
```diff
@@ -39,8 +39,9 @@ def get_items_yarn(self, tree, test):
                 items[unique_key] = item
             elif element.get("type") == "error":
                 error = element.get("data")
+                msg = "yarn audit report contains errors: %s"
                 raise ValueError(
-                    "yarn audit report contains errors: %s", error
+                    msg, error
```
Comment (cneill) [inline dojo/tools/yarn_audit/parser.py:44]: No need for a newline here...

**Review by cneill — COMMENTED** [review]
Just a few minor typos/style suggestions. Sorry to bug you with these, but linter PRs seem like a good opportunity to resolve them.

**Review by cneill — APPROVED** [review]

**Review by mtesauro — APPROVED** [review]
Approved

**Review by hblankenship — APPROVED** [review]

**Pull request merged**